stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/CallStatement.java

```diff
@@ -116,11 +118,21 @@ public Object execute() {
                 // returned result set or ref cursor OUT params we should cleanup the connection.
                 cleanupResources(resultSets, stmt, conn, !isInTransaction);
             }
-        } catch (Throwable e) {
+        } catch (SQLException e) {
+            cleanupResources(resultSets, stmt, conn, !isInTransaction);
+            // handleErrorOnTransaction(context);
+            // checkAndObserveSQLError(context, "execute stored procedure failed: " + e.getMessage());
+            return SQLDatasourceUtils.getSQLDatabaseError(e, "execute stored procedure failed: ");
+        } catch (DatabaseException e) {
+            cleanupResources(resultSets, stmt, conn, !isInTransaction);
+            // handleErrorOnTransaction(context);
+            // checkAndObserveSQLError(context, "execute stored procedure failed: " + e.getMessage());
+            return SQLDatasourceUtils.getSQLDatabaseError(e, "execute stored procedure failed: ");
```

Comment:
Isn't the logic in the two catch blocks duplicated?

Reply:
In both catch I'm calling getSQLDatabaseError method with different exception. So it is calling the overloaded getSQLDatabaseError method